homeassistant/components/overkiz/entity.py

```diff
@@ -0,0 +1,105 @@
+"""Parent class for every Overkiz device."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Callable
+
+from pyoverkiz.enums import OverkizAttribute, OverkizState
+from pyoverkiz.models import Device
+
+from homeassistant.components.sensor import SensorEntityDescription
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import OverkizDataUpdateCoordinator
+from .executor import OverkizExecutor
+
+
+class OverkizEntity(CoordinatorEntity):
+    """Representation of an Overkiz device entity."""
+
+    coordinator: OverkizDataUpdateCoordinator
+
+    def __init__(self, device_url: str, coordinator: OverkizDataUpdateCoordinator):
+        """Initialize the device."""
+        super().__init__(coordinator)
+        self.device_url = device_url
+        self.base_device_url, *_ = self.device_url.split("#")
+        self.executor = OverkizExecutor(device_url, coordinator)
+
+        self._attr_assumed_state = not self.device.states
+        self._attr_available = self.device.available
+        self._attr_name = self.device.label
+        self._attr_unique_id = self.device.device_url
+
+    @property
+    def device(self) -> Device:
+        """Return Overkiz device linked to this entity."""
+        return self.coordinator.data[self.device_url]
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return device registry information for this entity."""
+        # Some devices, such as the Smart Thermostat have several devices in one physical device,
+        # with same device url, terminated by '#' and a number.
+        # In this case, we use the base device url as the device identifier.
+        if "#" in self.device_url and not self.device_url.endswith("#1"):
+            # Only return the url of the base device, to inherit device name and model from parent device.
+            return {
+                "identifiers": {(DOMAIN, self.executor.base_device_url)},
```

Comment:
Does this meet the unique id requirements ?
https://developers.home-assistant.io/docs/entity_registry_index/#example-acceptable-sources-for-a-unique-id

Reply:
Yes, the device URL is the unique id used by Overkiz. The reason why we split it here is since Overkiz uses multiple subsystemids for the same physical device. In Home Assistant this would be seen as 1 device, with multiple entities. 

```
device URL (<protocol>://<gatewayId>/<deviceAddress>[#<subsystemId>])
```